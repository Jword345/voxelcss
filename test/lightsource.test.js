'use strict';

const LightSource = require('../lib/LightSource');

describe('Light Source', function() {
    let lightsource;
    let position = [1, 2, 3];
    beforeEach(function() {
        lightsource = new LightSource(...position);
    });
    it('can set travel distance', function() {
        expect(lightsource.getTravelDistance()).toMatchSnapshot();
        lightsource.setTravelDistance();
        expect(lightsource.getTravelDistance()).toMatchSnapshot();
        lightsource.setTravelDistance('Not a number');
        expect(lightsource.getTravelDistance()).toMatchSnapshot();
        lightsource.setTravelDistance(1000);
        expect(lightsource.getTravelDistance()).toMatchSnapshot();
    });
    it('can get and set brightness', function() {
        lightsource.setBrightness(1, 2);
        expect(lightsource.getBrightness()).toMatchSnapshot();
    });
});
