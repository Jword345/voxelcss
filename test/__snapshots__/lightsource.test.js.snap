// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Light Source can get and set brightness 1`] = `
Array [
  1,
  2,
]
`;

exports[`Light Source can set travel distance 1`] = `500`;

exports[`Light Source can set travel distance 2`] = `500`;

exports[`Light Source can set travel distance 3`] = `500`;

exports[`Light Source can set travel distance 4`] = `1000`;
